docs/book/routing.md

```diff
@@ -1,725 +1,65 @@
 ```
 
-The above would match the following:
-
-- `modules.laminas.dev` would dispatch the `Index` controller's `index`
-  action of the `Module` module.
-- `modules.ci.laminas.dev` would dispatch the `Index` controller's `index`
-  action of the `Module` module.
-- `packages.laminas.dev` would dispatch the `Index` controller's `index`
-  action of the `Package` module.
-- `packages.dev.laminas.dev` would dispatch the `Index` controller's
-  `index` action of the `Package` module.
-
-The `Url` controller plugin or view helper may be used to generate URLs
-following the above example:
+Note that when adding multiple routes, the last route in the list will be checked first.
 
-```php
-// reuse the route matched parameters to generate URLs
-echo $this->url('modules.laminas.dev/index', [], [], true);
-echo $this->url('packages.laminas.dev/index', [], [], true);
-```
+## HTTP Route Types
 
-> ### may_terminate and child_routes placement
->
-> When defining child routes pay attention that the `may_terminate` and
-> `child_routes` definitions are in same level as the `options` and `type`
-> definitions. A common pitfall is to have those two definitions nested in
-> `options`, which will not result in the desired routes.
+Laminas MVC ships with the following HTTP route types:
 
-## Console Route Types
+- [Hostname](https://docs.laminas.dev/laminas-router/routing/#laminasrouterhttphostname) matches domains and subdomains. Example: `docs.laminas.dev`.
+- [Literal](https://docs.laminas.dev/laminas-router/routing/#laminasrouterhttpliteral) matches an exact URI path. Example: `/contact-us`
+- [Method](https://docs.laminas.dev/laminas-router/routing/#laminasrouterhttpmethod) matches HTTP verbs. Example: `post,put` for a route that submits a form.
+- [Part](https://docs.laminas.dev/laminas-router/routing/#laminasrouterhttppart) allows crafting a tree of possible routes based on segments of
+  the URI path. Example: `/blog` can have child routes with `/rss` and `/subscribe`, which would match `/blog/rss` and `/blog/subscribe` respectively.
+- [Regex](https://docs.laminas.dev/laminas-router/routing/#laminasrouterhttpplaceholder), available since version 3.2.0 of the router, matches a URI path using a regular expression. Example: `/product/(?<id>[0-9]+)` would match `/product/001`, with the `id` parameter containing `001`.
+- [Scheme](https://docs.laminas.dev/laminas-router/routing/#laminasrouterhttpscheme) matches the URI scheme. Example: `https`.
+- [Segment](https://docs.laminas.dev/laminas-router/routing/#laminasrouterhttpsegment) matching one or more segments of a URI path. Example: `/:blog/:article` would match `/blog/why-use-mvc`, with the `article` parameter containing `why-use-mvc`.
+- `Query` is deprecated since version 2.3.0 of the router and removed in version 3.0.0. You can now add query parameters without a query route.
+- `Wildcard` is deprecated since version 2.3.0 of the router and removed in version 3.0.0. Use the `Segment` route type instead.
```

Comment:
The information about the availability of a route type should be covered in the laminas-router component, a short list is enough here on this page.
Deprecated types should no longer be used, so I would not list those here either.

Here is a preview of the current list:

![Bildschirmfoto 2022-02-05 um 11 22 36](https://user-images.githubusercontent.com/103362/152638200-e2fe8dd4-e0ec-49b8-b6cf-4eb165f339b6.png)

I think a simple line break would increase the overview:
![Bildschirmfoto 2022-02-05 um 11 25 42](https://user-images.githubusercontent.com/103362/152638308-ff944992-aef0-42c6-8f10-8d08baf3d9d7.png)

```suggestion
- [Hostname](https://docs.laminas.dev/laminas-router/routing/#laminasrouterhttphostname) matches domains and subdomains.  
  Example: `docs.laminas.dev`.
- [Literal](https://docs.laminas.dev/laminas-router/routing/#laminasrouterhttpliteral) matches an exact URI path.  
  Example: `/contact-us`
- [Method](https://docs.laminas.dev/laminas-router/routing/#laminasrouterhttpmethod) matches HTTP verbs.  
  Example: `post,put` for a route that submits a form.
- [Part](https://docs.laminas.dev/laminas-router/routing/#laminasrouterhttppart) allows crafting a tree of possible routes based on segments of
  the URI path.  
  Example: `/blog` can have child routes with `/rss` and `/subscribe`, which would match `/blog/rss` and `/blog/subscribe` respectively.
- [Regex](https://docs.laminas.dev/laminas-router/routing/#laminasrouterhttpplaceholder), matches a URI path using a regular expression.  
  Example: `/product/(?<id>[0-9]+)` would match `/product/001`, with the `id` parameter containing `001`.
- [Scheme](https://docs.laminas.dev/laminas-router/routing/#laminasrouterhttpscheme) matches the URI scheme.  
  Example: `https`.
- [Segment](https://docs.laminas.dev/laminas-router/routing/#laminasrouterhttpsegment) matching one or more segments of a URI path.  
  Example: `/:blog/:article` would match `/blog/why-use-mvc`, with the `article` parameter containing `why-use-mvc`.
```

Reply:
Another option can be a definition list:
![Bildschirmfoto 2022-02-05 um 11 28 40](https://user-images.githubusercontent.com/103362/152638379-1bb5ddfc-f3e3-4ab1-bb87-642479aac884.png)

```suggestion
[Hostname](https://docs.laminas.dev/laminas-router/routing/#laminasrouterhttphostname)
:   matches domains and subdomains.  
    Example: `docs.laminas.dev`.
[Literal](https://docs.laminas.dev/laminas-router/routing/#laminasrouterhttpliteral)
:   matches an exact URI path.  
    Example: `/contact-us`
[Method](https://docs.laminas.dev/laminas-router/routing/#laminasrouterhttpmethod)
:   matches HTTP verbs.  
    Example: `post,put` for a route that submits a form.
[Part](https://docs.laminas.dev/laminas-router/routing/#laminasrouterhttppart)
:   allows crafting a tree of possible routes based on segments of the URI path.  
    Example: `/blog` can have child routes with `/rss` and `/subscribe`, which would match `/blog/rss` and `/blog/subscribe` respectively.
[Regex](https://docs.laminas.dev/laminas-router/routing/#laminasrouterhttpplaceholder)
:   matches a URI path using a regular expression.
    Example: `/product/(?<id>[0-9]+)` would match `/product/001`, with the `id` parameter containing `001`.
[Scheme](https://docs.laminas.dev/laminas-router/routing/#laminasrouterhttpscheme)
:   matches the URI scheme.  
    Example: `https`.
[Segment](https://docs.laminas.dev/laminas-router/routing/#laminasrouterhttpsegment)
:   matching one or more segments of a URI path.  
    Example: `/:blog/:article` would match `/blog/why-use-mvc`, with the `article` parameter containing `why-use-mvc`.
```

See also: https://python-markdown.github.io/extensions/definition_lists/

Reply:
I went for option 2: definition list.